cmd/oras/root/repo/tags.go

```diff
@@ -115,10 +131,23 @@ func showTags(cmd *cobra.Command, opts *showTagsOptions) error {
 	}
 	err = finder.Tags(ctx, opts.last, func(tags []string) error {
 		for _, tag := range tags {
+			// if --oci-layout-path is used with a repository path, filter the
+			// tags under the repository.
+			if targetPrefix != "" {
+				if scopedTag, ok := strings.CutPrefix(tag, targetPrefix); ok {
+					tag = scopedTag
+				} else {
+					continue
+				}
```

Comment:
nit: code style: less cognitive load.
```suggestion
				scopedTag, ok := strings.CutPrefix(tag, targetPrefix)
				if !ok {
					continue
				}
				tag = scopedTag
```